src/connector/src/source/base.rs

```diff
@@ -149,9 +149,10 @@ pub struct SourceEnumeratorContext {
     pub connector_client: Option<ConnectorClient>,
 }
 
-#[derive(Clone, Copy, Debug, Default)]
+#[derive(Clone, Debug, Default)]
 pub struct SourceEnumeratorInfo {
     pub source_id: u32,
+    pub source: Option<Source>,
```

Comment:
Why we need to add this?

Reply:
For Iceberg, we need this `Source` to validate that the schema users provided is validated.

Reply:
Can we do that in the optimizer stage as other connectors do? During the `create source` process, we fetch all related info and do checks there